Libraries/Network/RCTNetworking.m

```diff
@@ -331,16 +340,47 @@ + (NSString *)decodeTextData:(NSData *)data fromResponse:(NSURLResponse *)respon
   // Attempt to decode text
   NSString *encodedResponse = [[NSString alloc] initWithData:data encoding:encoding];
   if (!encodedResponse && data.length) {
+    
+    
+    
     // We don't have an encoding, or the encoding is incorrect, so now we
     // try to guess (unfortunately, this feature is available in iOS 8+ only)
     if ([NSString respondsToSelector:@selector(stringEncodingForData:
                                                encodingOptions:
                                                convertedString:
                                                usedLossyConversion:)]) {
-      [NSString stringEncodingForData:data
-                      encodingOptions:nil
-                      convertedString:&encodedResponse
-                  usedLossyConversion:NULL];
+      
+      encoding = [NSString stringEncodingForData:data
+                                 encodingOptions:nil
+                                 convertedString:&encodedResponse
+                             usedLossyConversion:NULL];
+      
+      if (encoding > 0 && isIncremental) {
```

Comment:
Putting it in this nested block means we won't do anything if the server sends a content-encoding header, but the issue may happen here too, no?